core/src/main/java/org/apache/iceberg/rest/HTTPHeaders.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.immutables.value.Value;
+
+/**
+ * Represents a group of HTTP headers.
+ *
+ * <p>Header name comparison in this class is always case-insensitive, in accordance with RFC 2616.
+ *
+ * <p>This class exposes methods to convert to and from different representations such as maps and
+ * multimap, for easier access and manipulation – especially when dealing with multiple headers with
+ * the same name.
+ */
+@Value.Style(depluralize = true)
+@Value.Immutable
+@SuppressWarnings({"ImmutablesStyle", "SafeLoggingPropagation"})
+public interface HTTPHeaders {
+
+  HTTPHeaders EMPTY = of();
+
+  /** Returns all the header entries in this group. */
+  List<HTTPHeader> entries();
```

Comment:
Nit: shouldn't this be `Set<HTTPHeader>`?

Reply:
I feel like we should also annotate this as `@NotNull`

Reply:
I don't see a `@NotNull` annotation available in iceberg-core? Besides, the immutable generated class is annotated with `@AllParametersAreNonNullByDefault`.

Reply:
@danielcweeks `@NotNull` shouldn't be needed here, because everything not marked as `@Nullable` or Optional is implicitly not null with Immutables